python/ppml/src/bigdl/ppml/fl/nn/pytorch/aggregator.py

```diff
@@ -102,27 +102,38 @@ def put_client_data(self, client_id, data):
 
 
     def aggregate(self):
-        input, target = [], None        
-        for ndarray_map in self.client_data.values():
+        input, target = [], None
+        # to record the order of tensors with client ID
+        client_ids = []
+        for cid, ndarray_map in self.client_data.items():
             for k, v in ndarray_map.items():
                 if k == 'input':
                     input.append(torch.from_numpy(v))
+                    client_ids.append(cid)
```

Comment:
this only works if the order of client inputs are not important (which is true for your current examples), which is generally not the case for an arbitrary interaction layer.

Reply:
should we add a parameter in user request (specifying the order of this user in Tensor list), or use some pre-defined way like alphabetical order, or some other approaches to solve the order requirement?

Reply:
Maybe each client can have a client ID that determines their order

Reply:
An order support is added in [new commit](https://github.com/intel-analytics/BigDL/pull/4699/commits/74101b2bdbf68129d9edd328f1b8629f6d4cc802)